modules/xmpp/JingleSessionPC.js

```diff
@@ -1186,29 +1134,45 @@ JingleSessionPC.prototype.notifyMySSRCUpdate = function (old_sdp, new_sdp) {
 
     if (added && add) {
         logger.info("Sending source-add", add.tree());
-        this.connection.sendIQ(add,
-            function (res) {
-                logger.info('got add result', res);
-            },
-            function (err) {
-                logger.error('got add error', err);
-            }
-        );
+        this.connection.sendIQ(
+            add, null, this.newJingleErrorHandler(add), IQ_TIMEOUT);
     } else {
         logger.log('addition not necessary');
     }
 };
 
-JingleSessionPC.onJingleError = function (session, error)
-{
-    logger.error("Jingle error", error);
-}
+JingleSessionPC.prototype.newJingleErrorHandler = function(request, failureCb) {
+    return function (errResponse) {
```

Comment:
Because what we want to have as a handler is a callback function(error). Apart from "error" it also needs "request" as a variable which will not be provided as callback argument. Also it optionally calls failureCb which would not be provided as one of callback arguments.